src/components/wifi_scan.rs

```diff
@@ -171,56 +172,44 @@ impl WifiScan {
-                                n.signal = signal;
-                                n.mac = w.mac;
-                                let channel = w.channel.parse::<u8>().unwrap_or(0);
-                                n.channel = channel;
-                            }
-                        } else {
-                            wifi_nets.push(WifiInfo {
-                                time: now,
-                                ssid: w.ssid.clone(),
-                                channel: w.channel.parse::<u8>().unwrap_or(0),
-                                signal: w.signal_level.parse::<f32>().unwrap_or(-100.00),
-                                mac: w.mac.clone(),
-                                color: COLORS_NAMES[wifi_nets.len() % COLORS_NAMES.len()],
-                            });
+                    let wifi_nets: Vec<WifiInfo> = nets.iter().map(|w| {
+                        WifiInfo {
+                            time: now,
+                            ssid: w.ssid.to_string(),
+                            channel: w.channel.parse::<u8>().unwrap_or(0),
+                            signal: w.signal_level.parse::<f32>().unwrap_or(-100.00),
+                            mac: w.mac.to_string(),
+                            color: COLORS_NAMES[nets.len() % COLORS_NAMES.len()],
                         }
-                    }
+                    }).collect();
 
-                    let t_send = tx.send(Action::Scan(wifi_nets));
-                    match t_send {
-                        Ok(n) => (),
-                        Err(e) => (),
+                    if tx.send(Action::Scan(wifi_nets)).is_err() {
+                        // Handle error
                     }
                 }
-                Err(_e) => (),
-            };
+                Err(_) => {
+                    // Handle error
+                }
+            }
         });
     }
 
     fn parse_networks_data(&mut self, nets: &Vec<WifiInfo>) {
-        // -- clear signal values
-        self.wifis.iter_mut().for_each(|item| {
-            item.signal = 0.0;
-        });
-        // -- add or update wifi info
+        let mut wifi_map: HashMap<String, WifiInfo> = self.wifis.drain(..)
+            .map(|w| (w.ssid.clone(), w))
+            .collect();
+
         for w in nets {
-            if let Some(n) = self.wifis.iter_mut().find(|item| item.ssid == w.ssid) {
-                n.copy_values(w.clone());
+            if let Some(existing) = wifi_map.get_mut(&w.ssid) {
+                existing.copy_values(w.clone());
             } else {
-                self.wifis.push(w.clone());
+                wifi_map.insert(w.ssid.clone(), w.clone());
             }
         }
-        // -- sort wifi networks by it's signal strength
-        self.wifis
-            .sort_by(|a, b| b.signal.partial_cmp(&a.signal).unwrap());
+
+        self.wifis = wifi_map.into_values().collect();
+        self.wifis.sort_by(|a, b| b.signal.partial_cmp(&a.signal).unwrap());
```

Comment:
while wifi scanning is not working properly ... duplicated networks, wrong colors & indexes ... on what distro/OS do you working please ?